py/torch_tensorrt/dynamo/_tracer.py

```diff
@@ -115,6 +115,9 @@ def get_dynamic_shapes_args(mod: torch.nn.Module, inputs: Any) -> dict[str, Any]
     args = list(signature(mod.forward).parameters.keys())
     dynamic_shapes = {}
     for input, input_name in zip(inputs, args[: len(inputs)]):
+        # if input.name is not None, also not empty str, use the input.name
+        if input.name is not None and len(input.name) > 0 and input.name != input_name:
```

Comment:
what is happening here ?

Reply:
This is to handle the scenario, in case the forward signure paramater sequence is x, y,
And our InputSpec sequence is y, x, we honor the sequence user defined in the InputSpec:
```
class logical_or(nn.Module):
    def forward(self, x, y):
        return torch.ops.aten.logical_or.default(x, y)
input_specs = [
    Input(
        min_shape=(3,),
        opt_shape=(4,),
        max_shape=(6,),
        name="y",
    ),
    Input(
        min_shape=(2, 1),
        opt_shape=(3, 1),
        max_shape=(5, 1),
        name="x",
    ),
]

```

Reply:
I feel like we need to do spec matching via the pytree flattening scheme not names, it is not ergonomic if every input spec needs to be named

Reply:
So the idea is if I have a function input `Tuple[Tuple[Any], Dict[str, Any]]`, the spec collection `Tuple[Tuple[Input], Dict[str, Input]]` made from `arg_inputs` and `kwargs_inputs` and the forward function should flatten to the same ordering using pytorch's pytree system. See refit for an example

Reply:
I think in the above example, we should let Pytorch handle wrong ordering and we shouldn't be re-ordering for users (since we don't expect them to provide names). We could throw a warning instead if we detect incorrect ordering.
In PyTorch, there are two ways to provide dynamic shapes for the above example
```py
ep = torch.export.export(gm, (x, y), dynamic_shapes = {"x": {0: Dim(min=2, max=5)}, "y": {0: Dim(min=3, max=6)} }

# (or)
ep = torch.export.export(gm, (x, y), dynamic_shapes = ({0: Dim(min=2, max=5)},  {0: Dim(min=3, max=6)},)
```
In the second option, the names are not considered but the order is assumed to be correct (x, y) and PyTorch will throw an error (or pass) if users provide wrong order (y, x) based on the model. I believe our behavior should be the same as well.